packages/compiler-cli/ngcc/test/packages/build_marker_spec.ts

```
@@ -94,58 +94,49 @@ function restoreRealFileSystem() {
-      expect(checkMarkerFile(createEntryPoint('/node_modules/@angular/common'), 'fesm2015'))
+  describe('checkMarker', () => {
+    it('should return false if the marker property does not exist', () => {
+      expect(checkMarker(createEntryPoint('/node_modules/@angular/common'), 'fesm2015'))
           .toBe(false);
     });
 
-    it('should return true if the marker file exists and contains the correct version', () => {
-      writeFileSync(
-          '/node_modules/@angular/common/__modified_by_ngcc_for_fesm2015__', '0.0.0-PLACEHOLDER',
-          'utf8');
-      expect(checkMarkerFile(createEntryPoint('/node_modules/@angular/common'), 'fesm2015'))
-          .toBe(true);
+    it('should return true if the marker property exists and contains the correct version', () => {
+      const pkg = JSON.parse(readFileSync('/node_modules/@angular/common/package.json', 'utf8'));
+      pkg.__modified_by_ngcc__ = {fesm2015: '0.0.0-PLACEHOLDER'};
+      writeFileSync('/node_modules/@angular/common/package.json', JSON.stringify(pkg), 'utf8');
+      expect(checkMarker(createEntryPoint('/node_modules/@angular/common'), 'fesm2015')).toBe(true);
     });
 
-    it('should throw if the marker file exists but contains the wrong version', () => {
-      writeFileSync(
-          '/node_modules/@angular/common/__modified_by_ngcc_for_fesm2015__', 'WRONG_VERSION',
-          'utf8');
-      expect(() => checkMarkerFile(createEntryPoint('/node_modules/@angular/common'), 'fesm2015'))
+    it('should throw if the marker property exists but contains the wrong version', () => {
+      const pkg = JSON.parse(readFileSync('/node_modules/@angular/common/package.json', 'utf8'));
+      pkg.__modified_by_ngcc__ = {fesm2015: 'WRONG_VERSION'};
+      writeFileSync('/node_modules/@angular/common/package.json', JSON.stringify(pkg), 'utf8');
+      expect(() => checkMarker(createEntryPoint('/node_modules/@angular/common'), 'fesm2015'))
```

Is this error thrown up to the API user? How should it be handled?

---

Good question - I think for your purposes it should be caught by the webpack plugin and surfaced to the user via whatever error reporting method webpack uses, right?

---

I refactored the `mainNgcc()` function so that it does not swallow exceptions. The logging of these errors to the console is now done in the `ngcc-main.ts` command line script.

---

SGTM